Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -280,6 +278,37 @@ void ReflectometryReductionOne::init() {
                   Direction::Input);
 }
 
+//----------------------------------------------------------------------------------------------
+/** Validate the input properties.
+*/
+std::map<std::string, std::string> ReflectometryReductionOne::validateInputs() {
+
+  std::map<std::string, std::string> result;
+
+  MatrixWorkspace_sptr decWS = getProperty("DetectorEfficiencyCorrection");
+  MatrixWorkspace_sptr runWS = getProperty("InputWorkspace");
+
+  if (!decWS) {
+    return result;
+  }
+
+  if (decWS->getNumberHistograms() > 1 &&
+      decWS->getNumberHistograms() != runWS->getNumberHistograms()) {
+    result["DetectorEfficiencyCorrection"] =
+        "The number of spectra in the "
+        "Detector Efficiency Correction workspace must either be 1 or equal to "
+        "the number of spectra in the input workspace";
+  } else if (decWS->blocksize() > 1) {
+    result["DetectorEfficiencyCorrection"] =
```

Comment:
In addition, we also decided to keep the `RegionOfDirectBeam` input property after all. This also means that the `Multidetector?` and `Perform Direct Beam Normalization` stages in the workflow diagram are preserved.